pkg/cmd/releasenotes/options.go

```diff
@@ -0,0 +1,116 @@
+// Copyright (C) 2021 ScyllaDB
+
+package releasenotes
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"sort"
+
+	"github.com/scylladb/scylla-operator/pkg/genericclioptions"
+	githubql "github.com/shurcooL/githubv4"
+	"golang.org/x/oauth2"
+)
+
+type GenerateOptions struct {
+	genericclioptions.IOStreams
+
+	OrganizationName string
+	RepositoryName   string
+	RepositoryPath   string
+
+	GithubToken string
+
+	ReleaseName string
+	StartRef    string
+	EndRef      string
+
+	ghClient *githubql.Client
+}
+
+func NewGitGenerateOptions(streams genericclioptions.IOStreams) *GenerateOptions {
+	return &GenerateOptions{
+		IOStreams:        streams,
+		OrganizationName: "scylladb",
+		RepositoryName:   "scylla-operator",
+		RepositoryPath:   ".",
+	}
+}
+
+func (o GenerateOptions) Validate() error {
+	if o.ReleaseName == "" {
+		return errors.New("release name cannot be empty")
+	}
+	if o.StartRef == "" {
+		return errors.New("start ref cannot be empty")
+	}
+	if o.EndRef == "" {
+		return errors.New("end ref cannot be empty")
+	}
+
+	return nil
+}
+
+func (o *GenerateOptions) Complete(ctx context.Context) error {
+	var httpClient *http.Client
+	if o.GithubToken != "" {
+		ts := oauth2.StaticTokenSource(
+			&oauth2.Token{AccessToken: o.GithubToken},
+		)
+		httpClient = oauth2.NewClient(ctx, ts)
+	}
+
+	o.ghClient = githubql.NewClient(httpClient)
+
+	return nil
+}
+
+func (o *GenerateOptions) Run(ctx context.Context) error {
```

Comment:
ideally this would be located in a file named `releasenotes.go` or `releasenotes_run.go` as klog prints the file name `options.go` is pretty generic